packages/patterns/chat.tsx

```diff
@@ -66,31 +66,35 @@ export default recipe<LLMTestInput, LLMTestResult>(
             Clear Chat
           </ct-button>
 
-          <ul>
+          <ct-vscroll showScrollbar height="320px" fadeEdges snapToBottom>
             {chat.map((msg, idx) => {
               return (
-                <li key={idx}>
-                  <strong>{msg.role}:</strong> {msg.content}
-                </li>
+                <ct-chat-message
+                  key={idx}
+                  role={msg.role}
+                  content={msg.content}
+                />
               );
             })}
             {derive(llmResponse.pending, (pending) =>
```

Comment:
so I _think_ this could also work as

<ct-chat-message
  role="assistant"
  content={ifElse(llmResponse.partial, llmResponse.partial, "...")}
/>

(and no need to pay attention to pending here, as partial also goes through undefined -> partial -> full result)